File: Python/singleton/chocolate_boiler.py
class ChocolateBoiler:
    __empty: bool
    __boiled: bool
    __uniqueInstance = None

    def __init__(self):
        self.__empty = True
        self.__boiled = False

    @staticmethod
    def getinstance():
        if ChocolateBoiler.__uniqueInstance is None:
            print("Creating unique instance of Chocolate Boiler")
            ChocolateBoiler.__uniqueInstance = ChocolateBoiler()
        print("Returning instance of Chocolate Boiler")
        return ChocolateBoiler.__uniqueInstance

    def fill(self) -> None:
        if self.isEmpty():
            self.__empty = False
            self.__boiled = False

    def drain(self) -> None:
        if not self.isEmpty() and not self.isEmpty():
            self.__empty = True

    def boil(self) -> None:
        if not self.isEmpty() and not self.isBoiled():
            self.__boiled = True

    def isEmpty(self) -> bool:
        return self.__empty

    def isBoiled(self) -> bool:
        return self.__boiled

File: Golang/singleton/chocolate_boiler.go
package singleton

import "fmt"


type ChocolateBoiler struct {
	empty bool
	boiled bool
}

var uniqueInstance *ChocolateBoiler

func NewChocolateBoiler() *ChocolateBoiler {
	if uniqueInstance == nil {
		fmt.Println("Creating unique instance of Chocolate Boiler")
		uniqueInstance = &ChocolateBoiler{
			empty: true,
			boiled: false,
		}
	}
	fmt.Println("Returning instance of Chocolate Boiler")
	return uniqueInstance
}

func (cb *ChocolateBoiler) Fill() {
	if cb.IsEmpty() {
		cb.empty = false
		cb.boiled = false
	}
}

func (cb *ChocolateBoiler) Drain() {
	if !cb.IsEmpty() && !cb.IsBoiled() {
		cb.empty = true
	}
}

func (cb *ChocolateBoiler) Boil() {
	if !cb.IsEmpty() && !cb.IsBoiled(){
		cb.boiled = true
	} 
}

func (cb *ChocolateBoiler) IsEmpty() bool {
	return cb.empty
}

func(cb *ChocolateBoiler) IsBoiled() bool {
	return cb.boiled
}